hip-config.cmake.in

```diff
@@ -0,0 +1,170 @@
+      if(${CMAKE_FIND_PACKAGE_NAME}_FIND_REQUIRED)
+        set(cmake_fd_required_arg REQUIRED)
+      endif()
+      find_package(${dep} ${cmake_fd_version}
+          ${cmake_fd_exact_arg}
+          ${cmake_fd_quiet_arg}
+          ${cmake_fd_required_arg}
+      )
+      string(TOUPPER ${dep} cmake_dep_upper)
+      if (NOT ${dep}_FOUND AND NOT ${cmake_dep_upper}_FOUND)
+        set(${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE "${CMAKE_FIND_PACKAGE_NAME} could not be found because dependency ${dep} could not be found.")
+        set(${CMAKE_FIND_PACKAGE_NAME}_FOUND False)
+        return()
+      endif()
+      set(cmake_fd_version)
+      set(cmake_fd_required_arg)
+      set(cmake_fd_quiet_arg)
+      set(cmake_fd_exact_arg)
+    endif()
+  endmacro()
+endif()
+
+set(HIP_COMPILER "@HIP_COMPILER@")
+set(HIP_RUNTIME "@HIP_RUNTIME@")
+
+set_and_check( hip_INCLUDE_DIR "@PACKAGE_INCLUDE_INSTALL_DIR@" )
+set_and_check( hip_INCLUDE_DIRS "${hip_INCLUDE_DIR}" )
+set_and_check( hip_LIB_INSTALL_DIR "@PACKAGE_LIB_INSTALL_DIR@" )
+set_and_check( hip_BIN_INSTALL_DIR "@PACKAGE_BIN_INSTALL_DIR@" )
+
+set_and_check(hip_HIPCC_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipcc")
+set_and_check(hip_HIPCONFIG_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipconfig")
+
+if(HIP_COMPILER STREQUAL "clang")
+  if(CMAKE_CXX_COMPILER MATCHES ".*hipcc")
+    execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version
+                    OUTPUT_STRIP_TRAILING_WHITESPACE
+                    OUTPUT_VARIABLE HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT)
+    if(HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT MATCHES "InstalledDir:[\t\r\n][\t\r\n]*([^\t\r\n])")
+      set(HIP_CLANG_ROOT ${CMAKE_MATCH_1})
+    else()
+      set(HIP_CLANG_ROOT /opt/rocm/llvm)
+    endif()
+  else()
+    get_filename_component(HIP_CLANG_ROOT "${CMAKE_CXX_COMPILER}" PATH)
+    get_filename_component(HIP_CLANG_ROOT "${HIP_CLANG_ROOT}" PATH)
+  endif()
+  file(GLOB HIP_CLANG_INCLUDE_SEARCH_PATHS ${HIP_CLANG_ROOT}/lib/clang/*/include)
+  find_path(HIP_CLANG_INCLUDE_PATH stddef.h
+      HINTS
+          ${HIP_CLANG_INCLUDE_SEARCH_PATHS}
+      NO_DEFAULT_PATH)
+  find_dependency(AMDDeviceLibs)
+  set(AMDGPU_TARGETS "gfx900;gfx906" CACHE STRING "AMD GPU targets to compile for")
+  set(GPU_TARGETS "${AMDGPU_TARGETS}" CACHE STRING "GPU targets to compile for")
+else()
+  find_dependency(hcc)
+endif()
+
+find_dependency(amd_comgr)
+
+include( "${CMAKE_CURRENT_LIST_DIR}/hip-targets.cmake" )
+
+#If HIP isnot installed under ROCm, need this to find HSA assuming HSA is under ROCm
+if( DEFINED ENV{ROCM_PATH} )
+     set(ROCM_PATH "$ENV{ROCM_PATH}")
+endif()
+
+#get_filename_component cannot resolve the symlinks if called from /opt/rocm/lib/hip
+#and do three level up again
+get_filename_component(_DIR "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
+get_filename_component(_IMPORT_PREFIX "${_DIR}/../../../" REALPATH)
+
+#if HSA is not under ROCm then provide CMAKE_PREFIX_PATH=<HSA_PATH>
+find_path(HSA_HEADER hsa/hsa.h
+  PATHS
+    "${ROCM_PATH}/include"
```

Comment:
This should not be necessary if you use the relative path.

Reply:
What if user wants to use ROCM_PATH to specify a non-standard ROCm path?

Reply:
It seems unlikely since cmake doesn't use `ROCM_PATH` to search for dependencies. So users will still need to set `CMAKE_PREFIX_PATH` which makes `ROCM_PATH` unnecessary.

Reply:
Removed. Thanks.